app/src/org/commcare/engine/references/JavaHttpReference.java

```diff
@@ -35,7 +39,12 @@ public OutputStream getOutputStream() throws IOException {
 
     @Override
     public InputStream getStream() throws IOException {
-        return generator.simpleGet(uri).body().byteStream();
+        Response<ResponseBody> response = generator.simpleGet(uri);
+        if (response.isSuccessful()) {
+            return response.body().byteStream();
+        } else {
+            throw new RuntimeException(Localization.get("install.fail.error", Integer.toString(response.code())));
```

Comment:
Granted that we expect HTTP errors to happen fairly regularly is throwing a `RuntimeException` the best behavior? Can we throw this as an `IOException` and catch that ? 

Also, won't this crash the application?

Reply:
We are catching the generic Exception [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/UpdateTask.java#L125) so it doesn't crash the app though can definitely wrap it as `IOException` if it's more clean.

Reply:
Ah gotcha, nice. Yeah I think adding a block for `IOException` there handling this case would be preferable, but could be just my taste